core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/OnPrepareForwarder.java

```diff
@@ -0,0 +1,62 @@
+package com.novoda.noplayer.exoplayer.forwarder;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.PlaybackParameters;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.novoda.noplayer.PlayerState;
+import com.novoda.noplayer.listeners.PreparedListeners;
+
+class OnPrepareForwarder implements ExoPlayer.EventListener {
+
+    private final PreparedListeners preparedListeners;
+    private final PlayerState playerState;
+
+    OnPrepareForwarder(PreparedListeners preparedListeners, PlayerState playerState) {
+        this.preparedListeners = preparedListeners;
+        this.playerState = playerState;
+    }
+
+    @Override
+    public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+        if (isReady(playbackState)) {
+            preparedListeners.onPrepared(playerState);
+        }
+    }
+
+    private boolean isReady(int playbackState) {
+        return playbackState == ExoPlayer.STATE_READY;
+    }
+
+    @Override
+    public void onTimelineChanged(Timeline timeline, Object manifest) {
+        //TODO should we send ?
```

Comment:
Once again, again. We need to figure out the mapping between `ExoPlayer` and `no-player` for these additional events 😄